src/components/pages/users-page/users-page.tsx

```diff
@@ -227,134 +438,71 @@ export const UsersPage = () => {
-                  <TableRow>
-                    <TableCell className={classes.rolePermissionCell}>Manage Workspace</TableCell>
-                    <TableCell className={classes.rolePermissionIconCell}>
-                      {availableRoles[selectedRole].canManageWorkspace && (
-                        <CheckIcon />
-                      )}
-                    </TableCell>
-                  </TableRow>
-                  <TableRow>
-                    <TableCell className={classes.rolePermissionCell}>View Roster</TableCell>
-                    <TableCell className={classes.rolePermissionIconCell}>
-                      {availableRoles[selectedRole].canViewRoster && (
-                        <CheckIcon />
-                      )}
-                    </TableCell>
-                  </TableRow>
-                  <TableRow>
-                    <TableCell className={classes.rolePermissionCell}>View Muster Reports</TableCell>
-                    <TableCell className={classes.rolePermissionIconCell}>
-                      {availableRoles[selectedRole].canViewMuster && (
-                        <CheckIcon />
-                      )}
-                    </TableCell>
-                  </TableRow>
-
-                  <TableRow>
-                    <TableCell className={classes.rolePermissionCell}>View PII</TableCell>
-                    <TableCell className={classes.rolePermissionIconCell}>
-                      {availableRoles[selectedRole].canViewPII && (
-                        <CheckIcon />
-                      )}
-                    </TableCell>
-                  </TableRow>
-                </TableBody>
-              </Table>
-            </>
-          )}
-        </DialogContent>
-        <DialogActions className={classes.roleDialogActions}>
-          <Button disabled={formDisabled} variant="outlined" onClick={cancelRoleSelection} color="primary">
-            Cancel
-          </Button>
-          <ButtonWithSpinner
-            disabled={formDisabled}
-            onClick={acceptRoleSelection}
-            color="primary"
-            loading={finalizeApprovalLoading}
-          >
-            Finalize Approval
-          </ButtonWithSpinner>
-        </DialogActions>
-      </Dialog>
-      <Dialog
-        open={alert.open}
-        onClose={handleAlertClose}
-        aria-labelledby="alert-dialog-title"
-        aria-describedby="alert-dialog-description"
-      >
-        <DialogTitle id="alert-dialog-title">{alert.title}</DialogTitle>
-        <DialogContent>
-          <DialogContentText id="alert-dialog-description">
-            {alert.message}
-          </DialogContentText>
+              Yes, Remove User from Group
+            </ButtonWithSpinner>
+          </DialogActions>
         </DialogContent>
-        <DialogActions>
-          <Button onClick={handleAlertClose} autoFocus>
-            OK
-          </Button>
-        </DialogActions>
       </Dialog>
+      <RoleSelectionDialog
```

Comment:
Is there a way that these could be re-used?  If all of the logic is moved to the role selection dialog in a separate file, then you could do something similar to the other dialogs where you could indicate whether or not it's an edit or an add as a property, which might simplify it a bit.

Reply:
Yeah, I wanted to merge these but didn't for two reasons:
1) the "assign role dialog" task was going to touch them again
2) a larger conversation about embracing redux